NEWS.md

```diff
@@ -35,6 +35,11 @@ Standard library changes
 
 #### SparseArrays
 
+* `SparseMatrixCSC` and `SparseVector` are refactored to use accessor functions.  It is now
+  highly recommended to use accessor methods like `size`, `nzrange`, `rowvals`, `nonzeros`,
+  and `nonzeroinds` instead of directly using field (e.g., `A.m`) whose names are
+  implementation details ([#32953]).
```

Comment:
I don't think we can say that the fields and their types of `SparseMatrixCSC` are implementation details. Many algorithms require explicit modifications to e.g. the `colptr` and the usage of this is spread far and wide throughout the ecosystem and the fields are in fact documented. If people want to support `AbstractSparseMatrixCSC` then they would need generic code, but I really think we should try to keep existing code that currently works with `SparseMatrixCSC` to keep working. The current structure of `SparseMatrixCSC` is very standard and what I blieve the vast majority of users need.

Reply:
> Many algorithms require explicit modifications to e.g. the `colptr` and the usage of this is spread far and wide throughout the ecosystem and the fields are in fact documented.

Isn't `dosomething!(getcolptr(A))` just as efficient as `dosomething!(A.colptr)`?

> I really think we should try to keep existing code that currently works with `SparseMatrixCSC` to keep working

All `SparseMatrixCSC` fields are accessible through equivalent `AbstractSparseMatrixCSC` interface I proposed in #33054.  So I think you can write an efficient algorithm on `AbstractSparseMatrixCSC` which is as efficient as `SparseMatrixCSC`?  Can you comment on them in #33054?

(Sorry, I probably should have merged this to #33054. I thought #33039 may take sometime to be merged so it might make sense to write a NEWS/docs for its predecessor #32953 which is what this NEWS is all about.)

Reply:
Whether necessary or not, direct access to sparse matrix fields is both common and documented. Changing their names in a breaking fashion does not seem viable for 1.x — more attention should have been paid to these names/APIs before 1.0, but it’s too late for that now.

One way forward is to change the field names to better ones and use property overloading to keep code using the legacy names working.

Reply:
(For people finding this issue now, the discussion of the assessor API is happening in #33054 and the discussion of the future deprecation strategy is in #33056)